core/store/src/trie/shard_tries.rs

```diff
@@ -47,6 +47,15 @@ impl ShardTries {
         }
     }
 
+    pub fn add_new_shards(&mut self, shards: &Vec<ShardUId>) {
+        Arc::get_mut(&mut self.caches)
+            .unwrap()
+            .extend(Self::get_new_cache(&shards).as_ref().clone());
```

Comment:
Let's change `get_new_cache(shards: &Vec<ShardUId>)` to return an `HashMap<ShardUId, TrieCache>` without an `Arc` -- this cloning of data is superfluous. 

Wow, `get_new_cache` is written in a somewhat more elaborate way then it needs to be, here's a simpler version: 

```rust
    fn get_new_cache(shards: &Vec<ShardUId>) -> HashMap<ShardUId, TrieCache> {
        shards.iter().map(|&shard_id| (shard_id, TrieCache::new())).collect()
    }
```